libp2p/wire.nim

```diff
@@ -93,24 +163,30 @@ proc createStreamServer*[T](ma: MultiAddress,
                             bufferSize: int = DefaultStreamBufferSize,
                             child: StreamServer = nil,
                             init: TransportInitCallback = nil): StreamServer
-                            {.raises: [Defect, LPError, MaInvalidAddress].} =
+                            {.raises: [Defect, LPError, MaInvalidAddress, CatchableError].} =
   ## Create new TCP stream server which bounds to ``ma`` address.
   if not(RTRANSPMA.match(ma)):
     raise newException(MaInvalidAddress, "Incorrect or unsupported address!")
 
-  try:
-    return createStreamServer(
-      initTAddress(ma).tryGet(),
-      cbproc,
-      flags,
-      udata,
-      sock,
-      backlog,
-      bufferSize,
-      child,
-      init)
-  except CatchableError as exc:
-    raise newException(LPError, exc.msg)
+  let addressesResolved = waitFor resolveMAddresses(@[ma])
```

Comment:
We should never call `waitFor` from a `proc` that can be invoked from another `async` proc - the behaviour is undefined for the most part.

Reply:
Oh. Then I must break the `listen` apis to make them async

Reply:
How about having a sync `resolve` in addition to the async one?

Reply:
The problem with making things `async` is that it can introduce subtle concurrency issues that then lead to deadlocks and such, not that it would be the case here, but just something to keep in mind.

Reply:
> How about having a sync `resolve` in addition to the async one?

That's possible, the actual resolve async is ~65 LOC, so I would just need to duplicate that and remove all async on the duplicate.

That would not be very elegant though, and would have to be done for each resolver.

Or, just use the stock resolver for sync version, but that could bring weird discrepancies